partiql-parser/src/main/antlr/PartiQLParser.g4

```diff
@@ -11,14 +11,16 @@ options {
  *
  */
 
-root
-    : (EXPLAIN (PAREN_LEFT explainOption (COMMA explainOption)* PAREN_RIGHT)? )? statement;
+file
+    : statement (COLON_SEMI statement)* COLON_SEMI? EOF
+    ;
```

Comment:
```suggestion
statements
    : statement (COLON_SEMI statement)* COLON_SEMI? EOF
    ;
```

Reply:
Interesting that the last statement doesn't require a semi-colon. I see no semi-colon making sense for a single statement, but something is off about this succeeding..

```
select 1 + 1;
select 2 + 2;
select 3 + 3
```

Maybe multiple statements require each to have a semi-colon? The other way to look at it is like an implicit list that does not require trailing "commas" aka semi-colons

```
[ statement1; statement2; statement3 ]   -- reasonable
[ statement1; statement2; statement3; ]  -- reasonable
```

I'm not sure, but something seems off about not requiring a semi-colon on the last statement. Maybe that's just my experience with languages whose omission of a semi-colon on the last statement _means_ something different than when a semi-colon is present (rust implicit return)!

Reply:
Updated to enforce semi-colon for the last statement within a multi-statement block. We can always add support for it later.